lib/discourse_tagging.rb

```diff
@@ -355,9 +355,15 @@ def self.filter_allowed_tags(guardian, opts = {})
     term = opts[:term]
     if term.present?
       term = term.gsub("_", "\\_").downcase
-      builder.where("LOWER(name) LIKE :term")
       builder_params[:cleaned_term] = term
-      builder_params[:term] = "%#{term}%"
+
+      if opts[:match_term_on_starts_with]
```

Comment:
This is not a documented option in the method comment above. Could this be a "term_type" similar to condition types used by `TagHashtagDataSource.search`?